src/main/java/ObjectPool.java

```diff
@@ -0,0 +1,81 @@
+import vi.viredis.client.ViRedis;
+
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedTransferQueue;
+
+public abstract class ObjectPool<T> implements Pool<T> {
+
+    private int size;
+
+    private boolean shutdown;
+
+    private String host;
+    private int port;
+
+    private BlockingQueue objects;
+
+    public ObjectPool(int size, String host, int port) {
+        this.size = size;
+        this.host = host;
+        this.port = port;
+        shutdown = false;
+        init();
+    }
+
+    /*
+     * initiate the pool with fix size
+     */
+    private void init() {
+        objects = new LinkedTransferQueue();
+        for (int i = 0; i < size; i++) {
+            objects.add(createNew(host, port));
+        }
+    }
+
+    private T createNew(String host, int port) {
+        try {
+            return (T) new ViRedis(host, port);
```

Comment:
If <T> is any type other than ViRedis (or a subclass thereof) you will probably get an exception at one point or another.

Reply:
done!